tachyon_python_bindings/src/lib.rs

```diff
@@ -0,0 +1,330 @@
+        Ok(())
+    }
+
+    pub fn insert_float(&self, timestamp: Timestamp, value: f64) -> PyResult<()> {
+        let _dummy_lock = self
+            .dummy_lock
+            .lock()
+            .map_err(|_| WrappedTachyonErr::MiscErr(String::from("Failed to acquire lock!")))?;
+        let mut inserter = self
+            .ptr
+            .lock()
+            .map_err(|_| WrappedTachyonErr::MiscErr(String::from("Failed to acquire lock!")))?;
+        inserter
+            .insert_float64(timestamp, value)
+            .map_err(WrappedTachyonErr::TachyonErr)?;
+        Ok(())
+    }
+}
+
+#[pyclass(name = "QueryResultData")]
+#[derive(Clone)]
+pub enum WrappedQueryResultData {
+    SignedInteger(Vec<i64>),
+    UnsignedInteger(Vec<u64>),
+    Float(Vec<f64>),
+}
+
+impl From<QueryUtilsResultData> for WrappedQueryResultData {
+    fn from(data: QueryUtilsResultData) -> Self {
+        match data {
+            QueryUtilsResultData::Integer64(values) => Self::SignedInteger(values),
+            QueryUtilsResultData::UInteger64(values) => Self::UnsignedInteger(values),
+            QueryUtilsResultData::Float64(values) => Self::Float(values),
+        }
+    }
+}
+
+#[pyclass(name = "QueryResult")]
+pub struct WrappedQueryResult {
+    #[pyo3(get)]
+    pub return_type: WrappedReturnType,
+    #[pyo3(get)]
+    pub value_type: WrappedValueType,
+    timestamps: Option<Vec<Timestamp>>,
+    data: WrappedQueryResultData,
+}
+
+#[pymethods]
+impl WrappedQueryResult {
+    pub fn get(&self) -> PyResult<(Option<Vec<Timestamp>>, WrappedQueryResultData)> {
+        // TODO: Make this more efficient
+        Ok((self.timestamps.clone(), self.data.clone()))
+    }
+}
+
+/// SAFETY: A connection is only single-threaded
+#[pyclass(name = "Connection")]
+pub struct WrappedConnection {
+    dummy_lock: Arc<Mutex<()>>,
+    ptr: Arc<Mutex<Connection>>,
+}
+
+// SAFETY: TODO
+unsafe impl Send for WrappedConnection {}
+
+// SAFETY: TODO
+unsafe impl Sync for WrappedConnection {}
+
+#[pymethods]
+impl WrappedConnection {
+    #[new]
+    pub fn new(path: String) -> PyResult<Self> {
+        Ok(Self {
+            dummy_lock: Arc::new(Mutex::new(())),
+            ptr: Arc::new(Mutex::new(
+                Connection::new(path).map_err(WrappedTachyonErr::TachyonErr)?,
+            )),
+        })
+    }
+
+    pub fn create_stream(&self, name: String, value_type: WrappedValueType) -> PyResult<()> {
+        let _dummy_lock = self
+            .dummy_lock
+            .lock()
+            .map_err(|_| WrappedTachyonErr::MiscErr(String::from("Failed to acquire lock!")))?;
+        let mut conn = self
+            .ptr
+            .lock()
+            .map_err(|_| WrappedTachyonErr::MiscErr(String::from("Failed to acquire lock!")))?;
+        conn.create_stream(name, value_type.into())
+            .map_err(WrappedTachyonErr::TachyonErr)?;
+        Ok(())
+    }
+
+    pub fn delete_stream(&self, name: String) -> PyResult<()> {
```

Comment:
we have this?

Reply:
It's just a stub for the future. Currently it panics on the Rust side, which I think is fine since the implementation will be finished in Rick's PR